### PR TITLE
Added base class for datamodules and refactored

### PR DESCRIPTION
**Pull request by Sly1029**
Base class should do the heavy lifting for the constructor now. The parameters to the base class are a bit confusing but that should not be too much of an issue given it is not visible to any other modules. 

The three tests that were failing before in #19 are still failing but we can make this a separate issue. I think it might have to do with how randoms are treated in Linux and MacOS.

### CODE REVIEW

**Comment by Sly1029** [comment]
> Base class should do the heavy lifting for the constructor now. The parameters to the base class are a bit confusing but that should not be too much of an issue given it is not visible to any other modules.

File: torchfl/datamodules/cifar.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
 from torchvision import transforms
 from torchvision.datasets import CIFAR10, CIFAR100
 
+from .datamodule import BaseDataModule
```
Comment (vivekkhimani) [inline torchfl/datamodules/cifar.py:24]:
```suggestion
from torchfl.datamodules.base import BaseDataModule
```

File: torchfl/datamodules/emnist.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
 from torchvision import transforms
 from torchvision.datasets import EMNIST
 
+from .datamodule import BaseDataModule
```
Comment (vivekkhimani) [inline torchfl/datamodules/emnist.py:24]:
```suggestion
from torchfl.datamodules.base import BaseDataModule
```

File: torchfl/datamodules/fashionmnist.py [hunk 3]
```diff
@@ -18,13 +18,17 @@
 from torchvision import transforms
 from torchvision.datasets import FashionMNIST
 
+from .datamodule import BaseDataModule
```
Comment (vivekkhimani) [inline torchfl/datamodules/fashionmnist.py:21]:
```suggestion
from torchfl.datamodules.base import BaseDataModule
```

**Review by vivekkhimani — CHANGES_REQUESTED** [review]
Looks great overall! Left a few nits to replace the relative imports with an absolute one for best practice.

Also, can we rename ```torchfl/datamodules/datamodule.py``` to ```torchfl/datamodules/base.py``` to be consistent with other subpackages?

**Comment by vivekkhimani** [comment]
And yes, good catch on the tests. It certainly has to do with the random number generation on different OSs and should be non-blocking for this PR!

File: torchfl/datamodules/cifar.py (hunk 1 above)
Comment (Sly1029) [reply torchfl/datamodules/cifar.py]: Yep good call

**Review by vivekkhimani — APPROVED** [review]
LGTM! 🚀

**Pull request merged**